src/components/Input/SketchCanvas.js

```diff
@@ -0,0 +1,237 @@
+`
+
+const ToolBar = styled.div`
+  display: flex;
+  flex-direction: row;
+  padding: 3px;
+  padding-left: 7px;
+  padding-right: 7px;
+  justify-content: space-between;
+  user-select: none;
+`
+/**
+ * @param width number, width of the entire component
+ * @param height number, height of the entire component
+ * @param invalid boolean, indicates dissatisfaction with current value
+ * @param errorMsg string, indicates some error
+ * @param onChange (result: string) => void, triggers with result (SVG or Base64) on change. Has debouncing.
+ * @param customRef a ref directly to the canvas component
+ * @param exportAsBase64 boolean, onChange() will return a base64 string instead of svg if true
+ */
+export default ({ width, height, invalid, errorMsg, onChange, customRef, exportAsBase64 }) => {
+  const [strokeWidth, setStrokeWidth] = useState(4)
+  const [strokeColor, setStrokeColor] = useState('black')
+  const [showPicker, setShowPicker] = useState(false)
+  const canvasRef = useRef()
+
+  const [debounceTimerId, setDebounceTimerId] = useState(null)
+  const loadedRef = useRef(false)
+
+  useEffect(() => {
+    loadedRef.current = true
+    return () => {
+      loadedRef.current = false
+    }
+  }, [])
+
+  const handleChange = e => {
+    if (!onChange) return
+
+    if (debounceTimerId) clearTimeout(debounceTimerId)
+    const newTimerId = setTimeout(async () => {
+      if (!loadedRef.current) return
+
+      if (exportAsBase64) {
+        const newBase64 = await canvasRef.current.exportImage('jpeg')
+        onChange(newBase64)
+      } else {
+        const newSVG = await canvasRef.current.exportSvg()
+        onChange(newSVG)
+      }
+    }, 500)
+
+    setDebounceTimerId(newTimerId)
+  }
+
+  const handleClear = () => {
+    if (canvasRef.current) canvasRef.current.clearCanvas()
+  }
+
+  const handleUndo = () => {
+    if (canvasRef.current) canvasRef.current.undo()
+  }
+
+  const handleRedo = () => {
+    if (canvasRef.current) canvasRef.current.redo()
+  }
+
+  return (
+    <CanvasContainer
+      style={{
+        width: width,
+        height: height,
+        border: invalid ? '3px solid red' : '',
+      }}
+      onClick={() => {
+        setShowPicker(false)
+      }}
+    >
+      {errorMsg && <ErrorMessage style={{ width: width }}>{errorMsg}</ErrorMessage>}
+
+      <ReactSketchCanvas
+        width={'100%'}
+        height={'100%'}
+        ref={mergeRefs(canvasRef, customRef)}
+        strokeWidth={strokeWidth}
+        strokeColor={strokeColor}
+        onChange={handleChange}
+        style={{ border: 'none' }}
+      />
+
+      <ToolBar>
+        <LeftIcons>
+          <CanvasIcon
+            src={colorpickerIcon}
+            style={{
+              marginRight: 10,
+              filter: showPicker
+                ? 'invert(31%) sepia(79%) saturate(7412%) hue-rotate(241deg) brightness(84%) contrast(127%)'
+                : 'invert(72%) sepia(0%) saturate(237%) hue-rotate(137deg) brightness(86%) contrast(87%)',
```

Comment:
Just curious how you were able to generate these filter colours for the image icons?

Reply:
I used a random online codepen I found
https://codepen.io/sosuke/pen/Pjoqqp